internal/environment_writer.go

```diff
@@ -22,6 +25,14 @@ func (w EnvironmentWriter) Write(dir string, env map[string]string) error {
 	}
 
 	for key, value := range env {
+		// this regex checks that map keys contain valid env var name characters,
+		// per https://pubs.opengroup.org/onlinepubs/9699919799/
+		validEnvVarRegex := regexp.MustCompile(`^[a-zA-Z_]{1,}[a-zA-Z0-9_]*$`)
```

Comment:
Not a huge performance concern, but since the regexp doesn't change over the loop, you can pull it out.

```suggestion
	// this regex checks that map keys contain valid env var name characters,
	// per https://pubs.opengroup.org/onlinepubs/9699919799/
	validEnvVarRegex := regexp.MustCompile(`^[a-zA-Z_]{1,}[a-zA-Z0-9_]*$`)
	for key, value := range env {
```